### PR TITLE
chore: enhance app theme capability for prod edX theme/branding

### PR DESCRIPTION
**Pull request by farhan-arshad-dev**
## Description

- Enhance app theme capability for prod edX theme/branding
- Integrate Program config updates
- Theming/branding code improvements for light and dark modes

Jira: [LEARNER-9783](https://2u-internal.atlassian.net/browse/LEARNER-9783), [LEARNER-9874](https://2u-internal.atlassian.net/browse/LEARNER-9874)
Github issue: [[Android] Theming Customization Improvements](https://github.com/openedx/openedx-app-android/issues/192)

**Note:** No major changes in the OpenEdx Theme.

### CODE REVIEW

**Review by omerhabib26 — DISMISSED** [review]

**Review by k1rill — DISMISSED** [review]

**Comment by volodymyr-chekyrta** [comment]
@farhan-arshad-dev, I see TODO in the description. 
Could you please add some screenshots?

Also, I see that this PR is breaking the Open edX theme.

| Design | PR |
| ------------- | ------------- |
|image | image|

**Comment by farhan-arshad-dev** [comment]
> @farhan-arshad-dev, I see TODO in the description. Could you please add some screenshots?
> 
> Also, I see that this PR breaks the Open edX theme by using a secondary button style instead of the primary one.
> 
> Design	PR
> image	image

Hi @volodymyr-chekyrta 

I am currently addressing the design team's feedback. Soon I'll push more appropriate code, that will not break the Openedx theme.

**Comment by volodymyr-chekyrta** [comment]
> Hi @volodymyr-chekyrta
> 
> I am currently addressing the design team's feedback. Soon I'll push more appropriate code, that will not break the Openedx theme.

@farhan-arshad-dev Thank you! 👍

**Comment by farhan-arshad-dev** [comment]
@volodymyr-chekyrta PR is ready for another pass of review.

**Comment by volodymyr-chekyrta** [comment]
Incorrect button color
| develop | farhan_ar/LEARNER-9783 |
| ------------- | ------------- |
|image | image|

**Comment by volodymyr-chekyrta** [comment]
Incorrect text field and cursor color
| develop | farhan_ar/LEARNER-9783 |
| ------------- | ------------- |
|image | image|

**Comment by volodymyr-chekyrta** [comment]
Incorrect color
| develop | farhan_ar/LEARNER-9783 |
| ------------- | ------------- |
|image |image|

File: auth/src/main/java/org/openedx/auth/presentation/signin/compose/SignInView.kt [hunk 1]
```diff
@@ -264,15 +266,15 @@ private fun AuthForm(
...
         }
 
         if (state.showProgress) {
             CircularProgressIndicator(color = MaterialTheme.appColors.primary)
         } else {
-            OpenEdXButton(
+            OpenEdXSecondaryButton(
                 modifier = buttonWidth.testTag("btn_sign_in"),
                 text = stringResource(id = coreR.string.core_sign_in),
                 onClick = {
```
Comment (volodymyr-chekyrta) [inline auth/src/main/java/org/openedx/auth/presentation/signin/compose/SignInView.kt:280]: This solution causes a lot of confusion. Why do we use a Secondary Button for the main action on the screen?
Please do not introduce a new button type, use styling for the existing one.

**Comment by farhan-arshad-dev** [comment]
@volodymyr-chekyrta all the comments are addressed, and ready for another pass.

File: core/src/openedx/org/openedx/core/ui/theme/Colors.kt
Comment (volodymyr-chekyrta) [inline core/src/openedx/org/openedx/core/ui/theme/Colors.kt:1]: Please, delete unused `light_inactive_button_text` and `dark_inactive_button_text`

File: core/src/openedx/org/openedx/core/ui/theme/Colors.kt [hunk 2]
```diff
@@ -10,24 +10,37 @@ val light_background = Color.White
...
+
+val light_primary_button_background = Color(0xFF3C68FF)
+val light_primary_button_border = Color(0xFF97A5BB)
+val light_primary_button_text = Color.White
+val light_primary_button_bordered_text = Color(0xFF3C68FF)
+val light_secondary_button_background = Color(0xFF3C68FF)
+val light_secondary_button_text = Color.White
+val light_secondary_button_border = Color(0xFF97A5BB)
+val light_secondary_button_bordered_background = Color.White
+val light_secondary_button_bordered_text = Color(0xFF3C68FF)
```
Comment (volodymyr-chekyrta) [inline core/src/openedx/org/openedx/core/ui/theme/Colors.kt:42]: Could we reuse them this way?
```suggestion
val light_primary_button_background = Color(0xFF3C68FF)
val light_primary_button_border = Color(0xFF97A5BB)
val light_primary_button_text = Color.White
val light_primary_button_bordered_text = Color(0xFF3C68FF)

val light_secondary_button_background = light_primary_button_background
val light_secondary_button_text = light_primary_button_text
val light_secondary_button_border = light_primary_button_border
val light_secondary_button_bordered_background = Color.White
val light_secondary_button_bordered_text = light_primary_button_bordered_text
```

File: core/src/openedx/org/openedx/core/ui/theme/Colors.kt [hunk 3]
```diff
@@ -66,31 +80,45 @@ val dark_background = Color(0xFF19212F)
...
+val dark_primary_button_background = Color(0xFF5478F9)
+val dark_primary_button_text = Color.White
+val dark_primary_button_border = Color(0xFF4E5A70)
+val dark_primary_button_bordered_text = Color(0xFF5478F9)
+
+val dark_secondary_button_background = Color(0xFF5478F9)
+val dark_secondary_button_text = Color.White
+val dark_secondary_button_border = Color(0xFF4E5A70)
+val dark_secondary_button_bordered_background = Color(0xFF19212F)
+val dark_secondary_button_bordered_text = Color(0xFF5478F9)
```
Comment (volodymyr-chekyrta) [inline core/src/openedx/org/openedx/core/ui/theme/Colors.kt:110]: And here
```suggestion
val dark_primary_button_background = Color(0xFF5478F9)
val dark_primary_button_text = Color.White
val dark_primary_button_border = Color(0xFF4E5A70)
val dark_primary_button_bordered_text = Color(0xFF5478F9)

val dark_secondary_button_background = dark_primary_button_background
val dark_secondary_button_text = dark_primary_button_text
val dark_secondary_button_border = dark_primary_button_border
val dark_secondary_button_bordered_background = Color(0xFF19212F)
val dark_secondary_button_bordered_text = dark_primary_button_bordered_text
```

File: core/src/main/java/org/openedx/core/ui/theme/AppColors.kt [hunk 4]
```diff
@@ -19,9 +21,16 @@ data class AppColors(
...
 
-    val buttonBackground: Color,
-    val buttonSecondaryBackground: Color,
-    val buttonText: Color,
+    val primaryButtonBackground: Color,
+    val primaryButtonText: Color,
+    val primaryButtonBorder: Color,
+    val primaryButtonBorderedText: Color,
+
+    val secondaryButtonBackground: Color,
```
Comment (volodymyr-chekyrta) [inline core/src/main/java/org/openedx/core/ui/theme/AppColors.kt:29]: What about adding some comments for clarity?

```suggestion
    val primaryButtonBorderedText: Color,

    //Default secondary button is the same as the primary button.
    //But you can change it if your brand has 2 accent colors
    val secondaryButtonBackground: Color,
```

**Review by volodymyr-chekyrta — COMMENTED** [review]
We can go with this approach 👍 
Just a few comments and we are good to merge

File: core/src/main/java/org/openedx/core/ui/theme/AppColors.kt (hunk 4 above)
Comment (farhan-arshad-dev) [reply core/src/main/java/org/openedx/core/ui/theme/AppColors.kt:29]: @volodymyr-chekyrta 
What are your thoughts on this?
```
// The default secondary button styling is identical to the primary button styling.
// However, you can customize it if your brand utilizes two accent colors.
```

File: core/src/openedx/org/openedx/core/ui/theme/Colors.kt (hunk 2 above)
Comment (farhan-arshad-dev) [reply core/src/openedx/org/openedx/core/ui/theme/Colors.kt]: addressed.

File: core/src/openedx/org/openedx/core/ui/theme/Colors.kt (hunk 3 above)
Comment (farhan-arshad-dev) [reply core/src/openedx/org/openedx/core/ui/theme/Colors.kt]: addressed.

File: core/src/main/java/org/openedx/core/ui/theme/AppColors.kt (hunk 4 above)
Comment (farhan-arshad-dev) [reply core/src/main/java/org/openedx/core/ui/theme/AppColors.kt]: addressed.

Comment (volodymyr-chekyrta) [reply core/src/main/java/org/openedx/core/ui/theme/AppColors.kt]: Sounds great to me! 👍

**Review by volodymyr-chekyrta — APPROVED** [review]
LGTM 🚀

**Comment by volodymyr-chekyrta** [comment]
@farhan-arshad-dev, feel free to squash and merge this PR

**Pull request merged**